packages/create-content-sdk-app/src/templates/nextjs-app-router (beta)/src/app/[[...path]]/page.tsx

```diff
@@ -11,22 +12,24 @@ import Providers from 'src/Providers';
 import Bootstrap from 'src/Bootstrap';
 
 type PageProps = {
-  params: Promise<{ path?: string[] }>;
+  params: Promise<{ path?: string[]; [key: string]: string | string[] | undefined }>;
+  searchParams: Promise<{ [key: string]: string | string[] | undefined }>;
 };
 
-export default async function Page({ params }: PageProps) {
+export default async function Page({ params, searchParams }: PageProps) {
   const { path } = await params;
 
   // Set preview to false until preview mode is integrated
-  const preview = { enabled: false, data: {} };
+  // const preview = { enabled: false, data: {} };
 
   // Fetch the page data from Sitecore
   let page;
-  if (preview.enabled) {
-    if (isDesignLibraryPreviewData(preview.data)) {
-      page = await client.getDesignLibraryData(preview.data);
+if (isCSDKPreview(await headers())) {
```

Comment:
why don't you use `draftMode`? Using `headers` likely force page into dynamic rendering effectively killing SSG